### ISSUE TITLE
Unregister headless server

### ISSUE THREAD

**Issue by dszgit** [corrados/jamulus#168]
As far as I can tell, when running a headless server (no GUI) that is registered with a central server via -e flag, there is no way to tell the central server to unregister it, other than to kill the server and start a new one *with* the GUI, on the same host.  (In the GUI there is a check box to register or unregister the server. )  As a result, if I start a public (headless) server, then kill it and start it in non-public mode (without -e), it still appears in the server list.

There should be a way to get a headless server to unregister itself.  Either by killing it with, say, a SIGQUIT signal rather than SIGTERM, e.g.:

     pkill -QUIT Jamulus

or by starting a new server with some special commandline argument to unregister, like this:

     Jamulus -n -s --unregister jamulus.fischvolk.de:22224

Or maybe both.

**Comment by corrados** [comment]
We have a similar issue here:
https://github.com/corrados/jamulus/issues/130
I.e. if Jamulus is killed, it will not clean up things correctly.

**Comment by pljones** [comment]
I'm actually working on this.  I've done a first pass through the code based on a few bits and pieces and I've got a second reduction up pass to do, followed by a final clean up pass, then I can start testing it.  It should work on Windows and Linux (and, given MacOS is *nix-like enough, there too).

So be a while yet.  But getting there...

**Comment by pljones** [comment]
OK, #130 _should_ fix this, if you want to give it a go.

**Comment by pljones** [comment]
SIGINT and SIGTERM for the headless server, closing the Window on the GUI server.  All work here.

**Comment by corrados** [comment]
I have merged #195. Is this issue fixed with that merge? If yes, please close this Issue.

**Comment by pljones** [comment]
@dszgit can you confirm and close?

**Comment by Snayler** [comment]
I've tested this, it also fixes the issue with server being listed in both Central Servers. As soon as we change Central Server, it removes the server from the Central Server it was previously connected to.

**Comment by dszgit** [comment]
Something in this merge makes me unable to compile on my Debian stable (10.3) system.  I checked that it is the change in question (commit b60c3ca) that introduced the problem.  I'm attaching the output of the build.

I am able to compile on my laptop running Debian Testing.  But I'm behind a NAT and can't fully test the new feature right now.  I did run and then kill a public server, and it went away from the list; and when I started it again in private mode it did not reappear.  But I think that would have happened before too because the central server cannot ping my server behind the NAT.

[b60c3ca-output.txt](https://github.com/corrados/jamulus/files/4622198/b60c3ca-output.txt)

**Comment by pljones** [comment]
`/qt4/` Ah.  That's probably it.  I've got QT5 for Windows and Linux.

**Comment by dszgit** [comment]
Ah. I was going to say I just followed the instructions on the wiki, but I see that Volker just removed the one for generic Linux using qt4.  I'm not sure which packages I should install for Debian Stable (10.x).   I guess I'll try the instructions for Ubuntu and see what happens.

**Comment by dszgit** [comment]
OK, that seems to have worked.  I compiled successfully on my Debian 10 system.  
When I tried server, at first I could not get it to register with the central server without the GUI, though it said it registered, and I could connect to it by net address.  In the GUI it all worked as expected.  But when I exited the GUI server and started it headless (-s -n -e ...) it said it registered but it never showed up.   But then all of sudden it started working.  Now it seems to work as advertised:  When I start in public mode (-e) it appears in the list.  When I kill it and start in private mode, it doesn't.  I can't reproduce the original behavior, though I know what I saw!  Things like this give me heartburn.

I'll play around with it a bit more before closing the issue, just in case I can reproduce the weird behavior.

**Comment by dszgit** [comment]
Incidentally, when I say "all of a sudden it started working", I mean after restarting several times, with and without registering, and seeing the same behavior multiple times.  Then one time it all worked.

**Comment by dszgit** [comment]
OK, I've played around with the server on my Debian 10, and also managed to compile on another machine running Debian 9 (need to upgrade that!).  I'm still mystified about the behavior I saw initially.  But as far as I can see now, everything is behaving as advertised.  So I will go ahead and close this.   @pljones thank you for your efforts to make this work!
(I will also update the wiki to indicate the Ubuntu build prerequisites also work for Debian 9 and 10)
